src/components/Footer/functions/GraphFunctions.tsx

```diff
@@ -0,0 +1,87 @@
+import cytoscape from "cytoscape";
+
+export const generateGraphFromUserInput = (
+  cy: cytoscape.Core,
+  input: string
+) => {
+  const lines = input.split("|");
+  cy.elements().remove();
+
+  const nodeIds = new Map<string, string>();
+
+  lines.forEach((line, index) => {
+    const [source, target, weight] = line.split(" ");
+
+    if (!nodeIds.has(source)) {
+      const sourceNodeId = `node${nodeIds.size}`;
+      nodeIds.set(source, sourceNodeId);
+
+      cy.add({
+        group: "nodes",
+        data: { id: sourceNodeId, label: `${source}` },
+        position: { x: 0, y: 0 },
+      });
+    }
+
+    if (!nodeIds.has(target)) {
+      const targetNodeId = `node${nodeIds.size}`;
+      nodeIds.set(target, targetNodeId);
+
+      cy.add({
+        group: "nodes",
+        data: { id: targetNodeId, label: `${target}` },
+        position: { x: 0, y: 0 },
+      });
+    }
+
+    const verifyWeight = (weight: any) => {
+      const parsedWeight = parseInt(weight);
+      if (Number.isNaN(parseInt(weight))) {
+        return 0;
+      } else {
+        return Math.abs(parsedWeight);
+      }
+    };
+
+    const edgeId = `edge${index}`;
+    cy.add({
+      group: "edges",
+      data: {
+        id: edgeId,
+        source: nodeIds.get(source),
+        target: nodeIds.get(target),
+        weight: verifyWeight(`${weight}`),
+      },
+    });
+  });
+  return cy.layout({ name: "grid" }).run();
+};
+
+export const handleInputBatch = (
+  cy: cytoscape.Core,
+  generateGraphFromUserInput: (cy: cytoscape.Core, input: string) => void
+) => {
+  const userInput = prompt(
+    'Digite as informações para gerar o grafo (por exemplo, "0 2 3|0 1 2|1 2 3"):'
```

Comment:
Maybe this should be in english too to maintain consistency accross the system

Reply:
Resolvido!